files/external_node_v2.rb

```diff
@@ -112,6 +112,9 @@ def build_body(certname,filename)
   puppet_facts = YAML::load(facts.gsub(/\!ruby\/object.*$/,''))
   hostname     = puppet_facts['values']['fqdn'] || certname
   
+  require 'facter'
```

Comment:
It's possible that Facter can't be loaded in this script as it depends on which Ruby is being used and how the master is installed - e.g. in a PE or AIO environment, this script might be running under the system Ruby, which may not have Facter installed.

I'd consider using a `begin`/`rescue LoadError` block and skipping this step if it can't be loaded.  Perhaps even use `hostname` as a fallback..